src/pages/profile/index.tsx

```diff
@@ -6,16 +6,42 @@ import { useGitHubQuery } from "@/hooks";
 import Image from "next/image";
 import Link from "next/link";
 import { exportAsImage } from "@/utils";
+import GitHubCalendar from "react-github-calendar";
+import { useState } from "react";
+
+interface Activity {
+  date: string;
+  count: number;
+  level: 0 | 1 | 2 | 3 | 4;
+}
 
 export default function Profile() {
   const { data } = useGitHubQuery<UserProfile>(userProfileQuery);
+  const [checked, setChecked] = useState<boolean>(false);
 
   if (!data) return "Loading...";
 
+  const selectLastHalfYear = (contributions: Activity[]) => {
+    const currentYear = new Date().getFullYear();
+    const currentMonth = new Date().getMonth();
+    const shownMonths = 6;
+
+    return contributions.filter((activity: Activity) => {
+      const date = new Date(activity.date);
+      const monthOfDay = date.getMonth();
+
+      return (
+        date.getFullYear() === currentYear &&
+        monthOfDay > currentMonth - shownMonths &&
+        monthOfDay <= currentMonth
+      );
+    });
+  };
```

Comment:
This feels like it should be simplified somehow.
A hint, can we define a `startDate` that is 6 months from now, and then just make sure that each `activity.date` is after that?

Reply:
I defined it's.